### DISCUSSION TITLE
Questions about the handling of schale.gg due to the migration of beta channels and the secondary use of SchaleDB information

### DISCUSSION THREAD

**Discussion by 2288-256** [lonqie/SchaleDB#77] [Questions] [answered]
# Question1 About the domain when the beta channel is officially implemented

In the future, if beta channel is officially implemented, will the domain use schale.gg? Or will it be schaledb.com (or its subdomain)?

# Question2  About the structural change of the data file including student.json in the future

Is there a possibility of significantly changing the structure of data files used in SchaleDB in the future, including beta channels, in the future?
(I've already seen #23, #27, #32, etc., and I know that the format will be changed without notice, but I would like to know if there is a specific schedule due to the major update of this site in the future.)

# Question3 About the translation contents in each language folder of data

It is mentioned in #32

> The main exception would be translations that do not originate from the game as this is the work o F other contributors. For these you will need permission from the translator (please contact me via discord @lonqie to dis Cuss if it's the case)

Do translations that are not derived from the game also include jp's translation (data/jp/localization.json, etc.)?

# Question3 If I want to use a data file, where do I access it to get it?

I am currently building an iOS app using these SchaleDB data files.

I need to get student.json and stages.json, but which site should I get them from, github (raw.githubusercontent.com) or the website (schale.gg)?

(Currently, considering the load on the website side, we are getting it from the github.)

# Question4 About acquiring resources that are not published on github including voice

How is it best to get resources that are not published on github, including voices in the domain of static.schale.gg?

(Example: I want you to save the cache and avoid getting as much as possible, etc.)

I apologize for the many questions, but I would appreciate your answers.

(This question was created using translation, so there may be some nuance differences.)

**Comment by lonqie** [comment, marked as answer]
> In the future, if beta channel is officially implemented, will the domain use schale.gg? Or will it be schaledb.com (or its subdomain)?

A final decision hasn't been made regarding this, I'll confirm this when the site is approaching release candidate. There will be a period where both the current site (schale.gg) and the new version will coexist, so during this time both domains will be accessible.

> Is there a possibility of significantly changing the structure of data files used in SchaleDB in the future, including beta channels, in the future? (I've already seen #23, #27, #32, etc., and I know that the format will be changed without notice, but I would like to know if there is a specific schedule due to the major update of this site in the future.)

It's for this reason why I permit usage with this disclaimer. There is not a plan or schedule for changes to the format of the files, fields are added/removed/changed according to Schale DB's needs.

The updated site will indeed contain a significant number of changes, and more changes may be made between beta and the full release. Also, I am evaluating whether the current approach of loading large JSON data files can be phased out in the future.

> Do translations that are not derived from the game also include jp's translation (data/jp/localization.json, etc.)?

Yes, a lot of the Japanese translation in that file (mostly user interface) is provided by a contributor.

> I need to get student.json and stages.json, but which site should I get them from, github (raw.githubusercontent.com) or the website (schale.gg)?

It does not really matter which domain you use the file from as they are identical. If you are planning on loading (hotlinking) those files directly from your own web app or project, I wouldn't recommend it as updates may break them.

> How is it best to get resources that are not published on github, including voices in the domain of static.schale.gg?
> (Example: I want you to save the cache and avoid getting as much as possible, etc.)

Resources stored there are only available from one domain. If you are intending on loading them directly (hotlinking), depending on what you are doing you may run into issues as static.schale.gg has some cross-origin restrictions in place.
